### PR TITLE
ARC-1072 move the task filtering to getnext task to stop pollution the global …

### PR DESCRIPTION
**Pull request by joshkay10**
Moved the filtering of task types to the getnextask function so that it doesn't pollute global values.

### CODE REVIEW

File: src/sync/installation.ts [hunk 1]
```diff
@@ -62,7 +62,11 @@ export const sortedRepos = (repos: Repositories): [string, RepositoryData][] =>
 			new Date(a[1].repository?.updated_at || 0).getTime()
 	);
 
-const getNextTask = async (subscription: Subscription): Promise<Task | undefined> => {
+const getNextTask = async (subscription: Subscription, jiraHost: string): Promise<Task | undefined> => {
+
+	const includeBuildAndDeployments = await booleanFlag(BooleanFlags.BACKFILL_FOR_BUILDS_AND_DEPLOYMENTS, false, jiraHost);
+	const tasks = taskTypes.filter(task => includeBuildAndDeployments ? taskTypes : task !== "build" && task !== "deployment");
```
Comment (mboudreau) [inline src/sync/installation.ts:68]: filter need to return true or false.  I think the function you're looking for is `task => includeBuildAndDeployments || task !== "build" && task !== "deployment"`

File: src/sync/installation.ts [hunk 2]
```diff
@@ -62,7 +62,11 @@ export const sortedRepos = (repos: Repositories): [string, RepositoryData][] =>
 			new Date(a[1].repository?.updated_at || 0).getTime()
 	);
 
-const getNextTask = async (subscription: Subscription): Promise<Task | undefined> => {
+const getNextTask = async (subscription: Subscription, jiraHost: string): Promise<Task | undefined> => {
```
Comment (mboudreau) [inline src/sync/installation.ts:65]: instead of passing in the `jiraHost`, it might make more sense to pass in the boolean of `includeBuildAndDeployments`

Comment (joshkay10) [reply src/sync/installation.ts:65]: it makes more sense to pass the boolean, it just got awkward with line 
162-`} else if (!(await getNextTask(subscription, jiraHost))) {`
	
Could have FF evaluating in 2 places(above each getNext) but seemed convoluted to evaluate a flag that might not get called....

If I wanted to evaluate it once, it got pretty messy fast passing it all over the show.....

File: src/sync/installation.ts (hunk 1 above)
Comment (joshkay10) [reply src/sync/installation.ts]: 😮‍💨  - fixing

File: src/sync/installation.ts (hunk 2 above)
Comment (mboudreau) [reply src/sync/installation.ts:65]: yeah, fair.

**Review by mboudreau — APPROVED** [review]

**Pull request merged**